### PR TITLE
Fix task exiting

### PR DESCRIPTION
**Pull request by anton-bondarev**
Fix some problems during task exit. Some resources allocated on stack and can be use after task has been deleted.

### CODE REVIEW

File: src/kernel/thread/thread_sched_wait.c [hunk 1]
```diff
@@ -47,17 +47,26 @@ static void sched_wait_timeout_handler(struct sys_timer *timer, void *data) {
...
 #define SCHED_WAIT_TIMER_DEF(_tmr) \
 	struct sys_timer *_tmr;
 
 #define SCHED_WAIT_TIMER_INIT(_tmr, ...) \
-	timer_set(&_tmr, __VA_ARGS__)
+	timer_set(&_tmr, __VA_ARGS__);\
+	thread_wait_add(&(thread_self()->thread_wait), _tmr)
+
+#define SCHED_WAIT_TIMER_ADD(_tmr) \
+		thread_wait_add(&(thread_self()->thread_wait), _tmr)
```
Comment (Deryugin) [inline src/kernel/thread/thread_sched_wait.c]: Extra tab

File: src/kernel/thread/thread_sched_wait.c [hunk 2]
```diff
@@ -47,17 +47,26 @@ static void sched_wait_timeout_handler(struct sys_timer *timer, void *data) {
 #define SCHED_WAIT_TIMER_INIT(_tmr, ...) \
 	timer_init_start(&_tmr, __VA_ARGS__)
 
+#define SCHED_WAIT_TIMER_ADD(_tmr) \
+		thread_wait_add(&(thread_self()->thread_wait), &_tmr)
```
Comment (Deryugin) [inline src/kernel/thread/thread_sched_wait.c]: Extra tab

Comment (Deryugin) [inline src/kernel/thread/thread_sched_wait.c]: Here the second argument is `&_tmr`, but in the other branch of `#ifdef` it's `_tmr`, is it correct?

File: src/kernel/thread/thread_sched_wait.c [hunk 3]
```diff
@@ -77,6 +86,7 @@ int sched_wait_timeout(clock_t timeout, clock_t *remain) {
 		return res;
 	}
 
+	SCHED_WAIT_TIMER_ADD(tmr);
```
Comment (alexkalmuk) [inline src/kernel/thread/thread_sched_wait.c]: It seems you add `tmp` to the list here in `SCHED_WAIT_TIMER_ADD` and above in `SCHED_WAIT_TIMER_INIT`, they both contain `thread_wait_add(&(thread_self()->thread_wait), _tmr)`. It is normal?

Comment (anton-bondarev) [reply src/kernel/thread/thread_sched_wait.c]: One of them contain ` thread_wait_add(&(thread_self()->thread_wait_list), &_tmr)`
I want to remove implementation except allocate on stack and register for thread signals, but In the future

File: src/kernel/thread/thread_sched_wait.c [hunk 4]
```diff
@@ -72,11 +54,13 @@ int sched_wait_timeout(clock_t timeout, clock_t *remain) {
 	}
 
 	cur_time = clock();
+	SCHED_WAIT_TIMER_ADD(tmr);
 	if ((res = SCHED_WAIT_TIMER_INIT(tmr, TIMER_ONESHOT, jiffies2ms(timeout),
 			sched_wait_timeout_handler, schedee_get_current()))) {
 		return res;
 	}
 
+
```
Comment (alexkalmuk) [inline src/kernel/thread/thread_sched_wait.c]: Redundant empty line.

File: src/kernel/thread/thread_wait/thread_wait_stub.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#include <kernel/time/timer.h>
+
+struct thread_wait {
+
+};
+
+#define SCHED_WAIT_TIMER_DEF(_tmr) \
+	struct sys_timer *_tmr;
+
+#define SCHED_WAIT_TIMER_INIT(_tmr, flags, msec, handler, param) \
```
[the remark this answers is not in the document]
Comment (anton-bondarev) [reply src/kernel/thread/thread_wait/thread_wait_stub.h]: the stub is implementation of routines thread_wait_add & thread_wait_del
although they can be safely remove from macros

**Pull request merged**